app/structures/adjacency_matrix.py

```diff
@@ -103,3 +103,22 @@ def to_directed_adjacency_list(self):
 
     def to_adjacency_list(self):
         return self.to_directed_adjacency_list()
+
+    def iterative_pagerank(self) -> Dict[int, float]:
+        n = len(self.matrix)
+
+        P = np.zeros((n, n))
+        for i in range(n):
+            for j in range(n):
+                P[i][j] = 0.85 * self.matrix[i][j] / sum(self.matrix[i]) + 0.15 / n
+
+        pagerank = np.full(n, 1/n)
+
+        while True:
+            previous = pagerank.copy()
+            pagerank = pagerank.dot(P)
+
+            if np.linalg.norm(pagerank - previous) < 10**-5:
+                break
+
+        return {index: round(value, 4) for index, value in enumerate(pagerank)}
```

Comment:
python files (not only) should have empty new line at the end of file:
https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-a-source-file